packages/suite/src/middlewares/suite/metadataMiddleware.ts

```diff
@@ -12,19 +12,33 @@ const metadata =
             action.payload = api.dispatch(metadataActions.setAccountMetadataKey(action.payload));
         }
 
+        const prevState = api.getState().metadata.entities || [];
+
         // pass action
         next(action);
 
         switch (action.type) {
-            case SUITE.RECEIVE_AUTH_CONFIRM:
-                if (
-                    action.success &&
-                    api.getState().metadata.enabled &&
-                    action.payload.metadata.status === 'disabled'
-                ) {
-                    api.dispatch(metadataActions.init());
+            // if labelable entitities differ from previous state after discovery completed init metadata
+            // todo: what if I add a hidden wallet but don't run discovery?
+            case '@common/wallet-core/discovery/complete': {
+                const { device } = api.getState().suite;
+
+                if (api.getState().metadata.enabled && device?.state) {
+                    const nextState = api.dispatch(
+                        metadataActions.getLabelableEntitiesDescriptors(),
+                    );
+                    if (prevState.join('') !== nextState.join('')) {
+                        api.dispatch(metadataActions.init());
+                    } else {
+                        console.log('states are equal!');
```

Comment:
this log can be removed before merge, right?

Reply:
todo: passphrase